R/pathwayEnrich.R

```diff
@@ -0,0 +1,131 @@
+
+# pathwayEnrich.R
+#'
+#' \code{PathwayEnrichment} a function that perform a pathway enrichment analysis 
+#' Return a barplot that show the enriched pathway and its p value(fisher test) that is smaller than 
+#' the ajusted p value. The null hypothesis for each fisher test: the pathway
+#' is not enriched in the given gene set.
+#'
+#' @param geneSet A list of gene symbol in the system that you want to perform pathway
+#'  enrichment analysis.
+#'  
+#' @param method method of cutoff that the visualization plot method can be BH or 
+#' bonferroni.
+#' reactInfo
+#' @param reactInfo a dataframe map the reactome pathway id to the reference gene symbol 
+#' 
+#' @return a barplot that show the enriched pathway and its p value(fisher test) 
+#' that is smaller than the ajusted p value.
+#' 
+#' @import ggplot2
+#' @importFrom stats fisher.test
+#' @importFrom stats p.adjust
+#' @author {Yufei Yang} (aut)
+#' @examples
+#' # Picking sample gene list to find the pathway enrichment and p value
+#' # Call the symComp helper function to generate list
+#' sampleGeneList<-c("CREBBP","NR1H3","RORA","SREBF1","SMARCD3")
+#' reactInfo<- fetchData("ReactomeSym")
+#' result <- PathwayEnrichment(sampleGeneList,"BH",reactInfo)
+#'
+#' @export
+PathwayEnrichment<-function(geneSet,method,reactInfo){
+  
+  #fetch HGNC reference genes 
+  pathway_df<-data.frame(pathway=c(),pval=c())
+  unique_pw<-unique(reactInfo$REACTOME_ID)
+  #get data information from the helper function
+  
+  
+  fisher_matrix<-matrix(nrow = 2, ncol = 2)
+  
+  pwList<-c()
+  #get all pathways in the given gene set
+  for(gene in geneSet){
+    pws<-reactInfo$REACTOME_ID[reactInfo$HGNC==gene]
+    for(pw in pws){
+      if(!is.na(pw)){
+        pwList<-c(pwList,pw)
+      }
+      
+    }
+  }
+  
+  uniquePW<-unique(pwList)
+  #occurance list of pathway for given gene set initialized
+  occurenceGene<-list()
+  #occurance list of pathway for all human gene initialized
+  occurence<-list()
+  for(pw in uniquePW){
+    occurenceGene[[pw]]<-0
+    occurence[[pw]]<-0
+  }
+  #calculate value for the occurance list of pathway for the given gene set
+  for(gene in geneSet){
+    pws<-reactInfo$REACTOME_ID[reactInfo$HGNC==gene]
+    for(pw in pws){
+      if(!is.na(pw)){
+        occurenceGene[[pw]]<-occurenceGene[[pw]]+1
+      }
+      
+    }
+  }
+  enrichment<-data.frame(pathway=c(),pval=c(),adjustedPvalBon=c())
```

Comment:
I would personally add some newlines to between code and the next comment line for readability